pkg/network/tracer/tracer.go

```diff
@@ -161,8 +161,7 @@ func newTracer(cfg *config.Config) (_ *Tracer, reterr error) {
 	}()
 
 	if tr.bpfErrorsCollector = ebpftelemetry.NewEBPFErrorsCollector(); tr.bpfErrorsCollector != nil {
-		coretelemetry.GetCompatComponent().RegisterCollector(tr.bpfErrorsCollector)
-
+		coretelemetry.GetCompatComponent().(coretelemetry.PrometheusComponent).RegisterCollector(tr.bpfErrorsCollector)
```

Comment:
I feel like having to cast the interface to a specific impl goes against the "component way" of doing things...
And it also means that the implementations are not interchangeable